server/command.go

```diff
@@ -121,6 +125,90 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		}
 		p.postCommandResponse(args, str.String())
 		return &model.CommandResponse{}, nil
+	case "set_channel_welcome":
+		if len(parameters) == 0 {
+			p.postCommandResponse(args, "`set_channel_welcome` command requires the message to be provided")
+			return &model.CommandResponse{}, nil
+		}
+
+		if channelInfo, appErr := p.API.GetChannel(args.ChannelId); appErr != nil {
+			errMsg := fmt.Sprintf(
+				"error occured while checking the type of the chanelId `%s`: `%s`",
+				args.ChannelId, err)
+			p.postCommandResponse(args, errMsg)
+			return &model.CommandResponse{}, nil
+		} else if channelInfo.Type == model.CHANNEL_PRIVATE {
+			p.postCommandResponse(
+				args, "welcome messages are not supported for direct channels")
+			return &model.CommandResponse{}, nil
+		}
+
+		// strings.Fields will consume ALL whitespace, so plain re-joining of the
+		// parameters slice will not produce the same message
+		message := strings.SplitN(args.Command, "set_channel_welcome", 2)[1]
+
+		key := fmt.Sprintf("%s%s", args.ChannelId, WELCOMEBOT_CHANNEL_WELCOME_KEY)
+		if appErr := p.API.KVSet(key, []byte(message)); appErr != nil {
+			errMsg := fmt.Sprintf(
+				"error occured while storing the welcome message for the chanel: `%s`",
+				appErr)
+			p.postCommandResponse(args, errMsg)
+			return &model.CommandResponse{}, nil
+		}
+
+		p.postCommandResponse(
+			args,
+			fmt.Sprintf("stored the welcome message:\n%s", message),
+		)
+		return &model.CommandResponse{}, nil
+
+	case "get_channel_welcome":
+		if len(parameters) > 0 {
+			p.postCommandResponse(args, "`get_channel_welcome` command does not accept any extra parameters")
+			return &model.CommandResponse{}, nil
+		}
+
+		key := fmt.Sprintf("%s%s", args.ChannelId, WELCOMEBOT_CHANNEL_WELCOME_KEY)
+		if data, appErr := p.API.KVGet(key); appErr != nil {
+			errMsg := fmt.Sprintf(
+				"error occured while retrieving the welcome message for the chanel: `%s`",
+				appErr)
+			p.postCommandResponse(args, errMsg)
+		} else if data == nil {
+			p.postCommandResponse(args, "welcome message has not been set yet")
+		} else {
+			p.postCommandResponse(args, string(data))
```

Comment:
Maybe add an explanation before the message. Something like "welcome message is:\n"